Formula/xidel.rb

```diff
@@ -0,0 +1,77 @@
+class Xidel < Formula
+  desc "XPath/XQuery 3.0, JSONiq interpreter to extract data from HTML/XML/JSON"
+  homepage "https://www.videlibri.de/xidel.html"
+  license "GPL-3.0-or-later"
+
+  class TargetedResource < Resource
+    def initialize(*rest, **kw_rest)
+      @target = nil
+      super(*rest, **kw_rest)
+    end
+
+    def target(val = nil)
+      return @target if val.nil?
+
+      @target = val
+    end
+  end
+
+  stable do
+    url "https://github.com/benibela/xidel/releases/download/Xidel_0.9.8/xidel-0.9.8.src.tar.gz"
+    sha256 "72b5b1a2fc44a0a61831e268c45bc6a6c28e3533b5445151bfbdeaf1562af39c"
+  end
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+    regex(%r{href=.*?/tag/Xidel_(\d+(?:\.\d+)+)["' >]}i)
```

Comment:
Do we really need both `:github_latest` and a `regex` here? Why doesn't an appropriately chosen `regex` suffice?

Reply:
I based this off of `Formula/jq.rb`.

https://github.com/Homebrew/homebrew-core/blob/a17a702423a4605ad403e139f02d0977b4397c13/Formula/jq.rb#L17-L21

Reply:
That doesn't explain why it is needed here. Also, I think `jq`'s `livecheck` is wrong. #93921

Reply:
My guess, looking at the regex, is that jq (similarly to Xidel) has version tags that aren't just the number. The regex's beginning ensures that you are looking at a tag, not some other text on the page that matches `v\d+`, and the group captures `1.2.3` from `jq-v1.2.3`. Xidel is in a similar boat.

I'll note that `Homebrew/livecheck/strategy/github_latest.rb` says:
```rb
        # The default regex used to identify a version from a tag when a regex
        # isn't provided.
        DEFAULT_REGEX = %r{href=.*?/tag/v?(\d+(?:\.\d+)+)["' >]}i.freeze
```
So the complexity here is reasonable; I'd go patch `DEFAULT_REGEX` first to be simpler before tackling these packages' livecheck regexes.

`strategy :github_latest` should be safe to elide.
```rb
        # The `Regexp` used to determine if the strategy applies to the URL.
        URL_MATCH_REGEX = %r{
          ^https?://github\.com
          /(?:downloads/)?(?<username>[^/]+) # The GitHub username
          /(?<repository>[^/]+)              # The GitHub repository name
        }ix.freeze
```

Reply:
We should get rid of `:github_latest`. That gets rate-limited.

Reply:
So, should the whole `livecheck` block be removed, or should I leave it in until a future treewide removal of `github_latest`-based livechecks?

Reply:
If `brew livecheck` returns correct results with the `livecheck` block omitted you can remove it entirely. If it does not, then you should keep the existing `regex`.